src/tree/node_text.rs

```diff
@@ -339,16 +314,7 @@ mod inner {
             let len = self.len();
             let amt = end_byte_idx - start_byte_idx;
 
-            // The unsafe here is just used for efficiency.  This can be
-            // replaced with a safe call to `copy_within()` on the slice once
-            // that API is stabalized in the standard library.
-            unsafe {
-                ptr::copy(
-                    self.buffer.as_ptr().add(end_byte_idx),
-                    self.buffer.as_mut_ptr().add(start_byte_idx),
-                    len - end_byte_idx,
-                );
-            }
+            self.buffer.copy_within(end_byte_idx..len, start_byte_idx);
```

Comment:
We might be able to remove some the first two asserts here since they also happen in `copy_within`?

Reply:
I don't think `copy_within` has an equivalent of the first assert (it works fine regardless of the relative positions of the source and destination indices).

But we can get rid of `assert!(end_byte_idx <= self.len())`, yeah.  In fact, even the char boundary checks make that redundant, I think.

Reply:
The first check, `assert!(start_byte_idx <= end_byte_idx);` is checked by `slice::range(..)`

Reply:
I must be missing something.  I don't see where `slice::range()` would be called with `start_byte_idx` and `end_byte_idx` as arguments...?  Unless you mean it gets called higher up in the call stack by the callers of `remove_range()`?  But I don't think it's good to rely on callers to check the invariants.

(Again, I'm probably missing something.)